src/robot/running/bodyrunner.py

```diff
@@ -716,3 +734,14 @@ class NoLimit(WhileLimit):
 
     def __enter__(self):
         pass
+
+
+class LimitExceeded(ExecutionFailed):
+
+    def __init__(self, on_limit, message):
+        super().__init__(message)
+        self.on_limit = on_limit
+
+    @property
+    def on_limit_pass(self):
+        return self.on_limit == 'pass'
```

Comment:
Comparing strings here feels a bit unsafe. Possibly `on_limit_pass` should be given directly to `__init__` i.e. the usage would be `LimitExceeded(self.on_limit == 'pass', message)`.

Reply:
Not sure if it matters where the unsafe string comparison is?